### PR TITLE
Session Persistence

### PR DESCRIPTION
**Pull request by CristianRWolox**
## Summary

- Persist headers and current user.
- Load current user to store from asycstorage.
- Implement stream of authentication (mount or not login screen if user exists).

## Trello Card
https://trello.com/c/6CxsroW8/37-n-104-persistencia-de-sesi%C3%B3n

### CODE REVIEW

File: wbooks/src/app/screens/Login/index.tsx [hunk 1]
```diff
@@ -17,13 +18,25 @@ interface DataForm {
...
 const Login = () => {
   const { control, handleSubmit, errors } = useForm();
-  const navigation = useNavigation();
   const dispatch = useDispatch();
 
-  const onSubmit = ({ email, password }: DataForm) => {
-    dispatch(actionCreators.login(email, password));
-    navigation.navigate('Library');
-  };
+  useEffect(() => {
```
Comment (guidoprinc) [inline wbooks/src/app/screens/Login/index.tsx]: this should not be in login, this should be on the main screen to check if there is any loggedUser. Also all this actions should be defined on a service and then call them with a single function in the main scree of the app. And for main screen i mean the index that displays the whole app that has the provider, navigator, etc.

File: wbooks/src/redux/auth/actions.ts [hunk 2]
```diff
@@ -1,18 +1,35 @@
...
+import { createTypes, completeTypes, withPostSuccess } from 'redux-recompose';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import Reactotron from 'reactotron-react-native';
 
-const completedActions = completeTypes({ primaryActions: ['LOGIN'] });
+const completedActions = completeTypes({ primaryActions: ['LOGIN', 'SET_CURRENT_USER'] });
 
 export const actions = createTypes(completedActions, '@@AUTH');
 
-const target = 'user';
```
Comment (guidoprinc) [inline wbooks/src/redux/auth/actions.ts:8]: since you changed user, it should probably crash in each screen you were calling the reducer for the user data

Comment (guidoprinc) [reply wbooks/src/redux/auth/actions.ts:8]: correct them

**Review by guidoprinc — CHANGES_REQUESTED** [review]

File: wbooks/src/app/screens/Login/index.tsx (hunk 1 above)
Comment (CristianRWolox) [reply wbooks/src/app/screens/Login/index.tsx]: Done!

File: wbooks/src/redux/auth/actions.ts (hunk 2 above)
Comment (CristianRWolox) [reply wbooks/src/redux/auth/actions.ts:8]: I corrected them in this commit, I made sure that app doesn't crash because of this change.

**Review by guidoprinc — APPROVED** [review]

**Pull request merged**